DownFall/Views/BackpackView.swift

```diff
@@ -16,79 +16,226 @@ import SpriteKit
     
-    init(playableRect: CGRect) {
+    init(playableRect: CGRect, viewModel: TargetingViewModel, levelSize: Int) {
         self.playableRect = playableRect
+        self.boardSize = CGFloat(levelSize)
         //height and width set ups
         height = playableRect.height * Style.Backpack.heightCoefficient
-        let backPackHeight: CGFloat = 280
-        let backPackWidth = height * 2/3
+        
+        // get the view model
+        self.viewModel = viewModel
+        
+        //get the tile size
+        self.tileSize = 0.9 * (playableRect.width / CGFloat(levelSize))
+        
+        // view container
+        self.viewContainer = SKSpriteNode(texture: nil, color: .clear, size: playableRect.size)
+        self.viewContainer.zPosition = Precedence.foreground.rawValue
+        
+        // background view
+        self.background = SKSpriteNode(color: .foregroundBlue, size: CGSize(width: playableRect.width, height: height))//setting the background position
+        self.background.position = CGPoint.positionThis(background.frame, inBottomOf: self.viewContainer.frame)
+        
+        //targeting area
+        self.targetingArea = SKSpriteNode(color: .clear, size: CGSize(width: playableRect.width, height: playableRect.height))
+        self.targetingArea.position = self.viewContainer.frame.center
+        self.targetingArea.name = self.targetingAreaName
+        
+        //inventory area
+        self.inventoryArea = SKSpriteNode(texture: nil, color: .clear, size: CGSize(width: playableRect.width, height: height))
+        self.inventoryArea.position = CGPoint.positionThis(self.inventoryArea.frame, inBottomOf: self.viewContainer.frame)
+
+        // center target area reticles
+        let marginWidth = playableRect.width - CGFloat(tileSize * boardSize)
+        let marginHeight = playableRect.height - CGFloat(tileSize * boardSize)
+        let bottomLeftX = playableRect.minX + marginWidth/2 + tileSize/2
+        let bottomLeftY = playableRect.minY + marginHeight/2 + tileSize/2
+        self.bottomLeft = CGPoint(x: bottomLeftX, y: bottomLeftY)
         
         super.init(texture: nil, color: .clear, size: CGSize(width: playableRect.width, height: height))
+        // "bind" to to the view model
+        self.viewModel.updateCallback = self.updated
         
         // setting our frame size
-        self.size = CGSize(width: playableRect.width, height: height)
-        self.position = CGPoint.positionThis(self.frame, inBottomOf: playableRect)
+//        self.size = CGSize(width: playableRect.width, height: playableRect.height)
         
         // item and description areas
-        itemArea = SKSpriteNode(color: .clear, size: CGSize(width: playableRect.width - backPackWidth, height: height/2))
-        itemArea?.position = CGPoint.positionThis(itemArea?.frame, inBottomOf: self.frame, anchor: .left)
+        itemArea = SKSpriteNode(color: .clear, size: CGSize(width: playableRect.width * 3 / 4, height: height/2))
```

Comment:
It seems like we need a variety of functions on CGRect.  WE already created one for subtracting the bottom fifth.  Let's make some more there to make this way easier to modify